### PR TITLE
[RUNTIME][CLML] Fix for Softmax op for 4D tensors

### PR DESCRIPTION
**Pull request by krishnaraj36**
- Fixed the softmax layer for 4D tensors to support for NCHW and NHWC layout types.

- Enabled relevant test cases for softmax layer

### CODE REVIEW

**Comment by krishnaraj36** [comment]
@echuraev Can you please take a look and let me know your feedback.
@srkreddy1238

File: src/runtime/contrib/clml/clml_runtime.cc [hunk 1]
```diff
@@ -528,15 +529,22 @@ class CLMLRuntime : public JSONRuntimeBase {
...
-        this->layer_.in_placeholder.insert(
-            {nid, MakeCLMLTensorFromJSONNode(node, CL_TENSOR_LAYOUT_NCHW_QCOM, dtype, node_data,
-                                             shape)});
+        if (layout == CL_TENSOR_LAYOUT_OPTIMAL_QCOM) {
+          this->layer_.in_placeholder.insert(
+              {nid, MakeCLMLTensorFromJSONNode(node, CL_TENSOR_LAYOUT_NCHW_QCOM, dtype, node_data,
+                                               shape)});
+        } else {
+          this->layer_.in_placeholder.insert(
+              {nid, MakeCLMLTensorFromJSONNode(node, layout, dtype, node_data, shape)});
```
Comment (echuraev) [inline src/runtime/contrib/clml/clml_runtime.cc:546]: Is it necessary to add an `ICHECK` to check the layout or not?

**Review by echuraev — COMMENTED** [review]
Just a small question. In general LGTM. @srkreddy1238 could you please review this PR?

File: src/runtime/contrib/clml/clml_runtime.cc (hunk 1 above)
Comment (krishnaraj36) [reply src/runtime/contrib/clml/clml_runtime.cc:546]: This if else statement is to support other layout format CL_TENSOR_LAYOUT_NHWC_QCOM as well for input to the clml subgraph.

**Review by srkreddy1238 — APPROVED** [review]
LGTM.

**Pull request merged**